stac_fastapi/types/stac_fastapi/types/core.py

```diff
@@ -306,6 +306,13 @@ def _landing_page(
                     "href": urljoin(base_url, "search"),
                     "method": "POST",
                 },
+                {
```

Comment:
Is this always available? Maybe only expose it if the filter extension is enabled and there are actually any queryables available?

Reply:
Unfortunately not, the API routes are only added via the filter extension here: https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/extensions/stac_fastapi/extensions/core/filter/filter.py#L85

There is currently no way to register link objects to API objects like the landing page or collections. Not sure how to proceed here.

Reply:
That would be useful in general: filter, language, collection-search and potentially also sort and query all may add links.

Reply:
The relationship between the returned data structures (e.g. the landing page) and links has been a pain point for a while, and was one of the issues that derailed the original backend breakout over a year ago; see https://github.com/stac-utils/stac-fastapi/pull/472 for one of the PRs in that space.

I personally haven't given the issue too much direct thought, so prototype solutions are welcome @constantinius. N.B. I'm no longer being supported to help maintain this project, so unfortunately reviews might come at a slower pace as I'm currently doing them on personal time 🙇🏼.

Reply:
@constantinius looking at this again (after a while), it looks like you could check for the filter extension in `BaseCoreClient` and `AsyncBaseCoreClient`, e.g. in: https://github.com/stac-utils/stac-fastapi/blob/ded2283f74d4f5722dd251a45c448e842ae9f219/stac_fastapi/types/stac_fastapi/types/core.py#L354

There's a `extensions` attribute on the client.

Reply:
in titiler-pgstac we test to see if the endpoint exist https://github.com/stac-utils/titiler-pgstac/blob/ffe0b68c56eac9c2d5040d61a9022d4705807881/titiler/pgstac/factory.py#L1006-L1020